DPGAnalysis/SiStripTools/plugins/APVCyclePhaseProducerFromL1ABC.cc

```diff
@@ -186,30 +190,59 @@ void APVCyclePhaseProducerFromL1ABC::produce(edm::Event& iEvent, const edm::Even
   if (iEvent.run() >= _firstgoodrun) {
     Handle<L1AcceptBunchCrossingCollection> pIn;
     iEvent.getByToken(_l1abccollectionToken, pIn);
-
+    Handle<TCDSRecord> tcds_pIn;
+    iEvent.getByToken(_tcdsRecordToken, tcds_pIn);
+    bool useTCDS(tcds_pIn.isValid() && !_forceSCAL);
+    const auto& tcdsRecord = useTCDS ? *tcds_pIn.product() : NULL;
```

Comment:
Reference to `nullptr` is undefined behavior.

Reply:
When it is null it is not accessed (null and access are both regulated by useTCDS), I don't understand what's the problem

Reply:
Ah, you do mean auto& of NULL?

Reply:
It is undefined behavior according to the language, so the compiler is free to do literally anything (even if not today, it may tomorrow). If you can't follow the pattern
```cpp
if (tcds_pIn.isValid()) {
   const auto& tcdsRecord = *tcds_pIn;
}
```
please use a pointer instead of a reference
```cpp
const auto* tcdsRecrod = useTCDS ? tcds_pIn.product() : nullptr;
```

Reply:
Fun fact: it compiled without any issue

Reply:
Lot's of undefined behavior compile fine.